Export masks for  NSE-only constructs `.`, `J`, `patterns` and `measure` (#6125)
Rdatatable/data.table: NEWS.md

14. `setNumericRounding()` now invisibly returns the old rounding value instead of `NULL`, which is now consistent with similar behavior by `setwd()`, `options()`, etc. Thanks @MichaelChirico for the report and @joshhwuu for the fix.

> 15. `.`, `J`, `measure`, and `patterns` are now exported for use within `[` and `melt()` functions. Previously, only `.N`, `.I`, `.GRP`, `.GRPI`, `.SD`, and `:=` were exported. This change helps package developers avoid defining these constructs inside functions, thereby preventing CRAN NOTES and ensuring consistency in data.table usage. See [#5604](https://github.com/Rdatatable/data.table/issues/5604) and [#5277](https://github.com/Rdatatable/data.table/issues/5277). Thanks to @tdhock for the suggestions and @Nj221102 for implementing this improvement.

Reviewer: again, what is consistency? please either delete or clarify

Author: consistency here means that  the behaviour of `.`, `J`, `measure`, and `patterns` constructs after being exported is predictable and aligning with the previously exported constructs like .N, .I, .GRP, .GRPI, .SD, and :=, like before we had to define them inside their functions but didn't have to do that with others which were exported, so after this PR,  behaviour of all NSE constructs are consistent,  i think #5604 had the consistency tag for same reason.

Reviewer: great that makes sense, please edit the NEWS item to clarify that meaning of consistency.
how about "... are now exported for use withing [ and melt functions, for consistency with `.N` ... which were the only previous NSE exports."

Author: updated 👍

> 15. Extended support for non-standard evaluation (NSE) constructs within `[` and `melt()` functions. Now, constructs like `.`, `J`, `measure`, and `patterns` are exported, ensuring they are consistently available. Previously, only `.N`, `.I`, `.GRP`, `.GRPI`, `.SD`, and `:=` were exported. This update helps maintain consistency and functionality across your data.table usage, [#5604](https://github.com/Rdatatable/data.table/issues/5604) and [#5277](https://github.com/Rdatatable/data.table/issues/5277). Thanks @tdhock for suggestions and @Nj221102 for implementing this improvement.

Reviewer: What is "Extended support" ?? can this first sentence be deleted? If not, why not? What value does it add?
"constructs like" implies to me that there may be others that are exported, that are not listed here, so can that be deleted too please? ( to avoid confusion) 
Also, news items typically start with the function/object name, can you please edit? How about `.`, `J`, `measure`, and `patterns` are exported...

Author: done 👍 , if needed further changes plz suggest.

> 15. Extended support for non-standard evaluation (NSE) constructs within `[` and `melt()` functions. Now, constructs like `.`, `J`, `measure`, and `patterns` are exported, ensuring they are consistently available. Previously, only `.N`, `.I`, `.GRP`, `.GRPI`, `.SD`, and `:=` were exported. This update helps maintain consistency and functionality across your data.table usage, [#5604](https://github.com/Rdatatable/data.table/issues/5604) and [#5277](https://github.com/Rdatatable/data.table/issues/5277). Note: `measure` and `patterns` do not need additional masking as they are defined within `fmelt.R`, while the following constructs are masked as functions: `.`, `J`, `.N`, `.I`, `.GRP`, `.GRPI`, `.SD`, and `:=`. Thanks @tdhock for suggestions and @Nj221102 for implementing this improvement.

Reviewer: I find this confusing: Note: `measure` and `patterns` do not need additional masking as they are defined within `fmelt.R`, while the following constructs are masked as functions: `.`, `J`, `.N`, `.I`, `.GRP`, `.GRPI`, `.SD`, and `:=`

`.` and `J` and `:=` are functions but the others are not.
 
can it be deleted?  if not can you please explain why you need that / what value it adds?

Author: No lets delete that if its excessive and confusing, i only added these because i thought these are few highlights of the PR but yeah end user doesn't need to know this and will find it confusing.

deleting now.

14. `setNumericRounding()` now invisibly returns the old rounding value instead of `NULL`, which is now consistent with similar behavior by `setwd()`, `options()`, etc. Thanks @MichaelChirico for the report and @joshhwuu for the fix.

> 15. `.`, `J`, `measure`, and `patterns` are now exported for use within `[` and `melt()` functions, for consistency with`.N`, `.I`, `.GRP`, `.GRPI`, `.SD`, and `:=` which were the only previous NSE exports. This change helps package developers avoid defining these constructs inside functions, thereby preventing CRAN NOTES and ensuring consistency in data.table usage. See [#5604](https://github.com/Rdatatable/data.table/issues/5604) and [#5277](https://github.com/Rdatatable/data.table/issues/5277). Thanks to @tdhock for the suggestions and @Nj221102 for implementing this improvement.

Author: done :)

14. `setNumericRounding()` now invisibly returns the old rounding value instead of `NULL`, which is now consistent with similar behavior by `setwd()`, `options()`, etc. Thanks @MichaelChirico for the report and @joshhwuu for the fix.

> 15. `.`, `J`, `measure`, and `patterns` are now exported to support non-standard evaluation (NSE) constructs within `[` and `melt()` functions. Previously, only `.N`, `.I`, `.GRP`, `.GRPI`, `.SD`, and `:=` were exported. This update ensures that these constructs are consistently available, maintaining consistency and functionality across your data.table usage. See [#5604](https://github.com/Rdatatable/data.table/issues/5604) and [#5277](https://github.com/Rdatatable/data.table/issues/5277). Thanks to @tdhock for the suggestions and @Nj221102 for implementing this improvement.

Reviewer: I don't understand "This update ensures that these constructs are consistently available, maintaining consistency and functionality across your data.table usage." 
Isn't this relevant for package developers? Is it true that after merging this PR, a package function which uses `.` will no longer have to define it inside the function, to avoid a CRAN NOTE? That could be mentioned.

Author: Yes i agree, that merging this PR will enable package developers to not worry about CRAN NOTE while using these construct and i have changed the news item accordingly, I am sorry about taking this much reviewing for a news item but i think these reviews have made sure the news item is relevant to the users, thanks.

16. `print.data.table` gains new argument `show.indices` and option `datatable.show.indices` that allows the user to print a `data.table`'s indices as columns without having to modify the `data.table` itself. Thanks @MichaelChirico for the report and @joshhwuu for the PR.

> 17. `.`, `J`, `measure`, and `patterns` are now exported for use within `[` and `melt()`, for consistency with other NSE exports like `.N` and `:=`, [#5604](https://github.com/Rdatatable/data.table/issues/5604). Package developers can now import these names to avoid `R CMD check` `NOTE`s about these being undefined variables. Thanks to @MichaelChirico and @ylelkes for the suggestions and @Nj221102 for implementing.

Reviewer 2: ```suggestion
17. `.`, `J`, `measure`, and `patterns` are now exported for use within `[` and `melt()`, for consistency with other NSE exports like `.N` and `:=`, [#5604](https://github.com/Rdatatable/data.table/issues/5604). Package developers can now import these names to avoid `R CMD check` `NOTE`s about them being undefined variables. Thanks to @MichaelChirico and @ylelkes for the suggestions and @Nj221102 for implementing.
```

> 17. `.`, `J`, `measure`, and `patterns` are now exported for use within `[` and `melt()` functions, for consistency with`.N`, `.I`, `.GRP`, `.GRPI`, `.SD`, and `:=` which were the only previous NSE exports. This change means that package developers can simply import these names to avoid `R CMD check` `NOTE`s about these being undefined variables. See [#5604](https://github.com/Rdatatable/data.table/issues/5604) and [#5277](https://github.com/Rdatatable/data.table/issues/5277). Thanks to @tdhock for the suggestions and @Nj221102 for implementing.

Reviewer 2: ```suggestion
17. `.`, `J`, `measure`, and `patterns` are now exported for use within `[` and `melt()`, for consistency with other NSE exports like `.N` and `:=`, [#5604](https://github.com/Rdatatable/data.table/issues/5604). Package developers can now import these names to avoid `R CMD check` `NOTE`s about these being undefined variables. Thanks to @MichaelChirico and @ylelkes for the suggestions and @Nj221102 for implementing.
```

> 17. `.`, `J`, `measure`, and `patterns` are now exported for use within `[` and `melt()` functions, for consistency with`.N`, `.I`, `.GRP`, `.GRPI`, `.SD`, and `:=` which were the only previous NSE exports. This change means that package developers will no longer see CRAN NOTEs about these being undefined variables, and no longer need to provide dummy/NULL definitions of these in order to avoid such NOTEs. See [#5604](https://github.com/Rdatatable/data.table/issues/5604) and [#5277](https://github.com/Rdatatable/data.table/issues/5277). Thanks to @tdhock for the suggestions and @Nj221102 for implementing this improvement.

Reviewer 2: ```suggestion
17. `.`, `J`, `measure`, and `patterns` are now exported for use within `[` and `melt()` functions, for consistency with`.N`, `.I`, `.GRP`, `.GRPI`, `.SD`, and `:=` which were the only previous NSE exports. This change means that package developers can simply import these names to avoid `R CMD check` `NOTE`s about these being undefined variables. See [#5604](https://github.com/Rdatatable/data.table/issues/5604) and [#5277](https://github.com/Rdatatable/data.table/issues/5277). Thanks to @tdhock for the suggestions and @Nj221102 for implementing.
```
